radio/src/gui/colorlcd/radio_trainer.cpp

```diff
@@ -29,21 +29,30 @@ RadioTrainerPage::RadioTrainerPage():
 {
 }
 
-class NoEditNumberEdit : public NumberEdit {
-public:
-  NoEditNumberEdit(Window* parent, const rect_t& rect, int vmin, int vmax, std::function<int()> getValue, std::function<void(int)> setValue = nullptr, WindowFlags windowFlags = 0, LcdFlags textFlags = 0) :
-    NumberEdit(parent, rect, vmin, vmax, getValue, setValue, windowFlags, textFlags)
-  {
-  }
-  // Disable all events, show only
-#if defined(HARDWARE_TOUCH)
-  bool onTouchEnd(coord_t, coord_t) {
-    return true;
-  }
-#endif  
-  void onEvent(event_t event) {
-    FormField::onEvent(event);
+class StaticNumber : public StaticText
+{
+  public:
+    StaticNumber(Window * parent, const rect_t & rect, std::function<int()> getValue, WindowFlags windowFlags = 0, LcdFlags textFlags = 0) :
+    StaticText(parent, rect, "", windowFlags, textFlags),
+    _getValue(std::move(getValue))
+    {    
+    }
+
+  void paint(BitmapBuffer * dc) {
+    auto curval = _getValue();
+    char buf[20];
+    if(textFlags & PREC1)
+      snprintf(buf, sizeof(buf), "%d.%.1d", curval / 10, abs(curval % 10));
+    else if(textFlags & PREC2)
+      snprintf(buf, sizeof(buf), "%d.%.2d", curval / 100, abs(curval % 100));
+    else
+      snprintf(buf, sizeof(buf), "%d", curval);
```

Comment:
I hope this is what was intended by the PREC flags.. value at -704 for instance with PREC1 should display as -70.4 PREC2 should be -7.04?

Reply:
That's correct, isn't it @eshifri ?